cibyl/_data/schemas/config.json

```diff
@@ -0,0 +1,193 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$id": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "additionalProperties": false,
+  "properties": {
+    "environments": {
+      "type": "object",
+      "additionalProperties": false,
+      "patternProperties": {
+        "^.*$": {
+          "type": "object",
+          "additionalProperties": false,
+          "patternProperties": {
+            "^.*$": {
+              "type": "object",
+              "additionalProperties": false,
+              "properties": {
+                "system_type": {
+                  "type": "string",
+                  "enum": [
+                    "jenkins",
+                    "zuul"
+                  ]
+                },
+                "sources": {
+                  "type": "object",
+                  "additionalProperties": false,
+                  "patternProperties": {
+                    "^.*$": {
+                      "type": "object",
+                      "oneOf": [
+                        {
+                          "$ref": "#/$defs/jenkins"
+                        },
+                        {
+                          "$ref": "#/$defs/jjb"
+                        },
+                        {
+                          "$ref": "#/$defs/es"
+                        },
+                        {
+                          "$ref": "#/$defs/zuul"
+                        },
+                        {
+                          "$ref": "#/$defs/zuul.d"
+                        }
+                      ]
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+    "plugins": {
+      "type": "array",
+      "additionalItems": false,
+      "items": {
+        "type": "string"
+      }
+    }
+  },
+  "required": [
+    "environments"
+  ],
+  "$defs": {
+    "jenkins": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "driver": {
+          "type": "string",
+          "const": "jenkins"
+        },
+        "url": {
+          "type": "string"
+        },
+        "username": {
+          "type": "string"
+        },
+        "token": {
+          "type": "string"
+        },
+        "cert": {
```

Comment:
`cert` might also be a string with the path to a file containing certificates. Is it possible to define a Union type in the schema?

Reply:
Done.